Libraries/Components/FocusableWindows/FocusableWindows.windows.js

```diff
@@ -0,0 +1,331 @@
+            this._componentProps['style'] = componentStyle;
+          }
+        }
+      } 
+    }
+
+    render() {
+      return (
+        <WindowsControl
+          { ...this._focusableProps }
+          ref={ this._setFocusableRef }
+        >
+          <Component
+            { ...this._componentProps }
+            ref={ this._setComponentRef }
+          />
+        </WindowsControl>
+      );
+    }
+
+    _setComponentRef = (ref): void => {
+      let previousComponent = this._component;
+      this._component = ref;
+      if (this._isMounted && this.props.componentRef && this._component !== previousComponent) {
+        this.props.componentRef(this._component);
+      }
+    }
+
+    _setFocusableRef = (ref): void => {
+      this._focusable = ref;
+    }
+
+    getComponent() {
+      return this._component;
+    }
+
+    focus() {
+      if (this._focusable) {
+        UIManager.dispatchViewManagerCommand(
+          ReactNative.findNodeHandle(this._focusable),
+          UIManager.WindowsControl.Commands.focus,
+          null);
+      }
+    }
+
+    blur() {
+      if (this._focusable) {
+        UIManager.dispatchViewManagerCommand(
+          ReactNative.findNodeHandle(this._focusable),
+          UIManager.WindowsControl.Commands.blur,
+          null);
+      }
+    }
+
+    setNativeProps(nativeProps) {
+      let focusableProps = {};
+      let componentProps = {};
+
+      let atLeastOneFocusableProp = false;
+      let atLeastOneComponentProp = false;
+
+      if (nativeProps) {
+        for (const key in nativeProps) {
+          if (key in FocusableWindowsTemplate.focusablePropTypes) {
+            // Property supported by WindowsControl
+            focusableProps[key] = nativeProps[key];
+            atLeastOneFocusableProp = true;
+          } else if (key !== 'style') {
+            // Property supported by Component 
+            componentProps[key] = nativeProps[key];
+            atLeastOneComponentProp = true;
+          } else {
+            // Style case is special because it has to be split:
+            // - "transform" should go to WindowsControl
+            // - everything else goes to Component
+            // The difference (compared with normal property processing) is that no flattening/style id resolution is needed
+            const styles = nativeProps['style'];
+            if (styles) {
+              let focusableStyle = {};
+              let childStyle = {};
+              for (const styleName in styles) {
+                if (styleName === 'transform') {
+                  focusableStyle[styleName] = styles[styleName];
+                  atLeastOneFocusableProp = true;
+                } else {
+                  childStyle[styleName] = styles[styleName];
+                  atLeastOneComponentProp = true;
+                }
+              }
+              focusableProps['style'] = focusableStyle;
+              componentProps['style'] = childStyle;
+            }
+          }
+        }
```

Comment:
Any reason not to put this logic into a helper which returns focusableProps, componentProps, atLeastOneFocusableProp, and atLeastOneComponentProp? Then it can be called in _splitProps and setNativeProps

Reply:
I'm torn.. One does flattening, one doesn't.. also, how do I return all those, as an object?

---
In reply to: [160827379](https://github.com/Microsoft/react-native-windows/pull/1546#discussion_r160827379) [](ancestors = 160827379)

Reply:
Yes, I was suggesting to return an object.

I didn't notice the difference in flattening. That could be solved by the function taking a callback or a flag.

I'm also okay with leaving it as is.

---
In reply to: [160847784](https://github.com/Microsoft/react-native-windows/pull/1546#discussion_r160847784) [](ancestors = 160847784,160827379)